test/fake_http_server.cc

```diff
@@ -89,11 +89,18 @@ void FakeServer::SendStreamResponse(const std::string& path,
   stream_it->second.SendToAllQueues(response);
 }
 
-void FakeServer::TerminateAllStreams() {
+bool FakeServer::TerminateAllStreams(time::seconds timeout) {
   // Send sentinel (empty string) to all queues.
   for (auto& s : handler_.path_streams) {
     s.second.SendToAllQueues("");
   }
+
+  // Block until all queues removed.
+  bool success = true;
+  for (auto& s : handler_.path_streams) {
+    success = success && s.second.WaitUntilNoWatchers(timeout);
```

Comment:
Should this synchronize with the empty string being sent, to avoid the scenario that a reconnect could happen between the previous for loop, and this one?

Reply:
Good catch! Rather than trying to synchronize these actions, I changed the test to check the total number of connections received, which I think is clearer anyway.